python/paddle/trainer_config_helpers/networks.py

```diff
@@ -435,6 +435,86 @@ def __extend_list__(obj):
         input=tmp, stride=pool_stride, pool_size=pool_size, pool_type=pool_type)
 
 
+@wrap_name_default("separable_conv")
+def img_separable_conv(input,
+                       num_channels,
+                       num_out_channels,
+                       filter_size,
+                       stride=1,
+                       padding=0,
+                       depth_multiplier=1,
+                       act=None,
+                       bias_attr=None,
+                       param_attr=None,
+                       shared_bias=True,
+                       layer_type=None,
+                       name=None):
+    """
+    Separable Convolution.
+
+    The separable convolution module is consisted of a depthwise convolution
+    that acts separately on input channels, followed by a pointwise convolution
+    with 1*1 kernels that mixes channels. It is used for Xception:
+    https://arxiv.org/pdf/1610.02357.pdf
+
+    :param input: input layer.
+    :type input: LayerOutput
+    :param num_channels: the number of input channels.
+    :type num_channels: int
+    :param num_out_channels: the number of output channels.
+    :type num_out_channels: int
+    :param filter_size: the filter size for the depthwise convolution.
+    :type filter_size: int|tuple
+    :param stride: the stride size for the depthwise convolution.
+    :type stride: int|tuple
+    :param padding: the padding size for the depthwise convolution.
+    :type padding: int|tuple
+    :param depth_multiplier: the number of filter for one channel in the
+                             depthwize convolution.
+    :type depth_multiplier: int
+    :param act: the activation function for the output.
+    :type act: BaseActivation
+    :param bias_attr: see img_conv_layer for details.
+    :type bias_attr: ParameterAttribute
+    :param param_attr: see img_conv_layer for details.
+    :type param_attr: ParameterAttribute
+    :param shared_bias: see img_conv_layer for details.
+    :type shared_bias: bool
+    :param layer_type: see img_conv_layer for details.
+    :type layer_type: bool
+    :return: layer's output
+    :rtype: LayerOutput
+    """
+    __depthwise_conv__ = img_conv_layer(
+        name="%s_depthwise_conv" % name,
```

Comment:
If the user doesn't set the name, there must be bug.

Reply:
Depthwise separable convolution used in many models(https://github.com/PaddlePaddle/Mobile/blob/develop/models/standard_network/mobilenet.py). But It seems that it usually has a batch norm layer after every conv, how about set an option ?

Reply:
Thank you very much for your reviews.

1. the `@wrap_name_default("separable_conv")` at the very beginning ensures the name will always be set even the users do not provide.

2. I surveyed the separable conv in TF and MXNet, there is no batch-norm between depthwise conv and pointwise conv. The batch norm comes after each of the separable_conv. The demo code you provided also contains no batch-nom between depthwise and pointwise conv:

![2017-12-15 15 45 59](https://user-images.githubusercontent.com/32189592/34032014-1d43e5da-e1af-11e7-849a-fef23d7693bc.png)